config/initializers/vault.rb

```diff
@@ -46,5 +45,38 @@ def initialize
         raise "Failed to get auth token from vault server"
       end
     end
+
+    def read(key)
+      Vault.address = get_host
```

Comment:
- check that this is threadsafe (uses Thread.current in the gem) or we this needs to live in Thread.exclusive
 - do not change global config and then leave it like that

```Ruby
def with_host
  old = Vault.address
  Vault.address = host
  yield
ensure
  Vault.address = old
end
```